pkg/controller/endpoint/trigger_time_tracker.go

```diff
@@ -99,16 +99,16 @@ func (t *TriggerTimeTracker) ComputeEndpointsLastChangeTriggerTime(
 	// minChangedTriggerTime is the min trigger time of all trigger times that have changed since the
 	// last sync.
 	var minChangedTriggerTime time.Time
-	// TODO(mm4tt): If memory allocation / GC performance impact of recreating map in every call
-	// turns out to be too expensive, we should consider rewriting this to reuse the existing map.
-	podTriggerTimes := make(map[string]time.Time)
 	for _, pod := range pods {
 		if podTriggerTime := getPodTriggerTime(pod); !podTriggerTime.IsZero() {
-			podTriggerTimes[pod.Name] = podTriggerTime
+			if state.lastPodTriggerTimes == nil {
+				state.lastPodTriggerTimes = make(map[string]time.Time)
```

Comment:
This will make the lastPodTriggerTimes map grow indefinitely over time, as the deleted pods won't be removed from the map anymore. 
IIRC, this was the main reason for recreating the map each time.

Reply:
How about adding a check of the length of state.lastPodTriggerTimes ?
If the size is bigger than 200, start from empty map.

Reply:
I'm no a big fan of using "magic number" in the code, 200 might change a lot in smaller clusters, but will probably end up being no-op in bigger ones. 
Probably a better idea would be to recreate the map every N calls. 

But may I ask, what is the reason behind making this change? Is recreating this map in every call causing some performance issues?

Reply:
We can wait till the related code is shown to be bottleneck.